cda-core/src/diag_kernel/ecumanager.rs

```diff
@@ -1574,7 +1619,37 @@ impl EcuManager {
                             let mapped_data = diag_type.type_.apply(&uds_data);
                             Ok(Some(mapped_data))
                         }
-                        datatypes::DataOperationVariant::EndOfPdu(_end_of_pdu_dop) => todo!(),
+                        datatypes::DataOperationVariant::EndOfPdu(end_of_pdu_dop) => {
+                            let structure =
+                                self.get_basic_structure(end_of_pdu_dop.field.basic_structure)?;
+
+                            if !value.is_array() {
+                                return Err(DiagServiceError::InvalidRequest(
+                                    "value should be an array".to_owned(),
```

Comment:
question: IIrc we discussed this on another PR a couple of days ago: Should error messages always start with a capital letter?
If we want this it should be made part of the code style. 
(I sent you the link to the discussion)

Reply:
yes I think we should, however, due to another change the error message in question wasn't necessary any more :S